### PR TITLE
Implement Online Datasource

### PR DESCRIPTION
**Pull request by morris-jason**
Fixes 5606, 5715, and 5570.

Supports Online Export/Import using DataSource and DataSink, overriding the default behavior of the toolbox's ArchiveFileHandler. Unpack and Diff are leveraged using DiskLabels in the tether.

### CODE REVIEW

File: lib/apiservers/engine/backends/archive.go [hunk 1]
```diff
@@ -248,46 +284,6 @@ func NewArchiveStreamWriterMap(mounts []types.MountPoint, dest string) *ArchiveS
 	return writerMap
 }
 
-// NewArchiveStreamReaderMap creates a new ArchiveStreamReaderMap.  After the call, it contains
```
Comment (morris-jason) [inline lib/apiservers/engine/backends/archive.go]: diff bug.

File: lib/apiservers/engine/backends/archive.go [hunk 2]
```diff
@@ -500,72 +546,6 @@ func (rm *ArchiveStreamReaderMap) FindArchiveReaders(containerSourcePath string)
 	return nodes, nil
 }
 
-func (rm *ArchiveStreamReaderMap) buildFilterSpec(containerSourcePath string, nodes []*ArchiveReader, startingNode *ArchiveReader) error {
```
Comment (morris-jason) [inline lib/apiservers/engine/backends/archive.go]: diff bug

File: lib/portlayer/storage/data_source.go [hunk 3]
```diff
@@ -53,6 +54,46 @@ func (m *MountDataSource) Source() interface{} {
 }
 
 // Export reads data from the associated data source and returns it as a tar archive
+func (m *MountDataSource) Stat(op trace.Operation, spec *archive.FilterSpec) (*FileStat, error) {
```
Comment (hickeng) [inline lib/portlayer/storage/data_source.go]: If we're passing in a FilterSpec as the input, we should return a `map[string]*FileStat` for the result - while it's fine for an implementation to return an error stating that it cannot deal with multiple inputs the signature should be consistent between intput/output to allow for other implementation decisions.

Comment (morris-jason) [inline lib/portlayer/storage/data_source.go]: Ok. Should I return 422 then if there's more than one path in filterspec Inclusions?

**Pull request closed without merging**